tests/io/test_readwrite.py

```diff
@@ -238,6 +238,18 @@ def test_replace_transformation_on_disk_non_raster(self, shapes, points):
                 t1 = get_transformation(SpatialData.read(f).__getattribute__(k)[elem_name])
                 assert type(t1) == Scale
 
+    def test_overwrite_files_without_backed_data(self, full_sdata):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            f = os.path.join(tmpdir, "data.zarr")
+            old_data = SpatialData()
+            old_data.write(f)
+            # Since not backed, no risk of overwriting backing data. Should not raise.
+            try:
+                full_sdata.write(f, overwrite=True)
+            except Exception as e:
+                assert "The file path specified is the same as the one used for backing" not in str(e)
+                raise  # Let other unexpected exceptions bubble up
```

Comment:
```suggestion
            full_sdata.write(f, overwrite=True)
```
What is the reason for the `try` `except` construct here? It is expected that this would not throw an error and if it does the test should fail. This would work already this way without the `try` `except` no? I understand that in this case you specifically want to test overwriting without backed data and only fail if this does not work, but I am just not certain whether it is worth it to write this block for it instead of just `full_sdata.write(f, overwrite=True)` in which case it could throw all errors which would be useful information in any case.

Reply:
I agree that the try/except is redundant. I used it to be more explicit about which exception of the previous behavior should not occur anymore by design (versus other unexpected exceptions).